### PR TITLE
Reader: remove duplicate request when locating feed by URL/domain

### PR DESCRIPTION
**Pull request by tyxla**
I'm currently working on creating more discoverable URLs for the Reader - `/read/feed/example.com` in addition to just `/read/feed/1234567`. 

Most of the work is done at the public APIs, however I'm occuring a small issue in Calypso.

### The problem 

When accessing a Reader feed URL by domain instead of ID, a duplicate API request is being done. The first request queries the API by domain, and the second one queries it by its ID. The second request is not necessary and should be removed. 

### How to reproduce

Try accessing this URL: http://calypso.localhost:3000/read/feeds/staff.tumblr.com

and you'll notice 2 identical requests to the API, made to the following API URLs:

* https://public-api.wordpress.com/rest/v1.1/read/feed/staff.tumblr.com?http_envelope=1&meta=site
* https://public-api.wordpress.com/rest/v1.1/read/feed/8932341?http_envelope=1&meta=site

Now try accessing this URL: http://calypso.localhost:3000/read/feeds/8932341

and you'll only notice a single request to:

https://public-api.wordpress.com/rest/v1.1/read/feed/8932341?http_envelope=1&meta=site

### Reason

I've investigated the issue and it appears it is a Calypso issue with handling domains in the feed store. 
To explain better what causes the issue, I think it is best if I describe it step-by-step:

* We load the following URL: https://wordpress.com/read/feeds/staff.tumblr.com
* Calypso's router matches `/read/feeds/:feed_id`, and dispatches the request to `feedListing()` (defined at `/client/reader/controller.js:220`).
* This creates a `FeedStream` instance, setting the `staff.tumblr.com` domain as a `feed_ID`.
* `FeedStream` uses a `FeedStore` instance to retrieve the feed by that domain.
* `FeedStore` replies and returns the new feed object, dispatching it back to the `FeedStream`.
* `FeedStream` receives new props, because the `feed_ID` is now the numeric ID and not the domain. Setting the new props triggers a state update and a rerender.
* `FeedStore` now attempts to fetch the feed by the numeric `feed_ID`. It does not exist in the `FeedStore`, so a new request is being done, even though it will always return the same feed data.

In a nutshell, the issue is that `FeedStore` does not recognize that the domain and its numeric ID correspond to the same feed.

### What this PR does

Once we retrieve the feed by domain, in the `FeedStore` we also set that same feed to the feed numeric ID. 

This way the `FeedStore` will immediately return the proper feed object when retrieving by numeric `feed_ID`, without the need to perform a new request.

### Testing

1) Go to: http://calypso.localhost:3000/read/feeds/staff.tumblr.com

2) An API request to: https://public-api.wordpress.com/rest/v1.1/read/feed/8932341?http_envelope=1&meta=site should no longer be triggered. 

3) An API request to https://public-api.wordpress.com/rest/v1.1/read/feed/staff.tumblr.com?http_envelope=1&meta=site is perfectly fine and is expected.

### CODE REVIEW

File: client/lib/feed-store/index.js [hunk 1]
```diff
@@ -30,6 +30,11 @@ var feeds = {},
 function setFeed( feedId, feed ) {
 	if ( feed !== feeds[ feedId ] ) {
 		feeds[ feedId ] = feed;
+
+		if ( feedId !== feed.feed_ID ) {
```
Comment (blowery) [inline client/lib/feed-store/index.js]: how was this called such that feed.feed_ID !== feedId? We should be normalizing at the call site, not in here.

**Comment by tyxla** [comment]
This was occurring with the previous implementation of the `/read/feed/?url=$url` endpoint. With the new endpoint approach this is no longer occurring, so this PR is no longer relevant. Closing.
